rpc/client.go

```diff
@@ -122,7 +146,7 @@ func NewClient(addr net.Addr, opts *retry.Options, context *Context) *Client {
 			close(c.Ready)
```

Comment:
the `Ready` channel loses its functionality with this change because it'll only work in the intended way until the client first encounters connection issues. You'll need to replace it when the peer goes dead and close it on each successful attempt. Plus access to it must be behind the lock, so it needs to become private.